config/samples/flows_v1alpha1_flowcollector.yaml

```diff
@@ -4,12 +4,19 @@ metadata:
   name: cluster
 spec:
   namespace: netobserv
+  deploymentType: DIRECT
```

Comment:
The name `deploymentType` sounds like a bit ambiguous to me. Maybe `communication`?

Reply:
I'm not fan of `deploymentType`, but maybe not `communication` either .. let's get more ideas :)

Reply:
what about `optionnalComponents` where we can list `Kafka` but also `Prometheus` and even `Loki` in the future if it become optional ? 
It will be very flexible and would allow any new options.

If we really want to be explicit, we can simply use `components` with values like `Agent-Processor` and `Agent-Kafka-Processor`

Reply:
Some ideas: `wiring`, `connect`, `messaging`, `linking`...

Reply:
`transfer`...